Automattic/wp-calypso
# Jetpack Backup: add learn more links to update credentials errors

Resolves https://github.com/Automattic/jetpack-backup-team/issues/490

Some error messages when updating credentials had a `Get help` button in the code, that were displayed as part of an error notice. In 2021 the error messages were switched from an error notice to an inline message description in https://github.com/Automattic/wp-calypso/pull/48106.

After some internal discussions, we agreed that we should add a `Learn more` links to point users to our support documentation in case they have any error, so the idea of this PR is to replace the unused `Get help` buttons with `Learn more` links, similar to what we did in https://github.com/Automattic/wp-calypso/pull/90409.

This is the error code + support link mapping suggested:

| Error code | Support link |
|---|---|
| service_unavailable | https://jetpack.com/support/backup/adding-credentials-to-jetpack/ |
| invalid_wordpress_path | https://jetpack.com/support/backup/adding-credentials-to-jetpack/#troubleshoot-remote-server-credentials-errors |
| read_only_install | https://jetpack.com/support/backup/ssh-sftp-and-ftp-credentials/#file-access-permission |
| helper_upload_failed | https://jetpack.com/support/backup/ssh-sftp-and-ftp-credentials/#file-access-permission |

## Proposed Changes

* Replace `Get Help` buttons with a `Learn more` links as part of error messages when updating credentials using the error code + support link mapping provided above.
* Add `calypso_jetpack_settings_credentials_error_learn_more_click` Tracks event for each link, passing the error code as a property.

| Error code | Screenshot |
|---|---|
| service_unavailable | ![CleanShot 2024-05-14 at 15 55 56@2x](https://github.com/Automattic/wp-calypso/assets/1488641/4ecfd37c-d72d-480e-96c4-27281e6a17ee) |
| invalid_wordpress_path | ![CleanShot 2024-05-14 at 15 53 54@2x](https://github.com/Automattic/wp-calypso/assets/1488641/545dd135-24d9-461b-b3ce-443a29b30be0) |
| read_only_install | ![CleanShot 2024-05-14 at 15 55 03@2x](https://github.com/Automattic/wp-calypso/assets/1488641/7fa52ba8-b84d-4043-b064-7b48495098ae) |
| helper_upload_failed | ![CleanShot 2024-05-14 at 15 52 16@2x](https://github.com/Automattic/wp-calypso/assets/1488641/c48a6ec0-841b-4d10-b6f1-eed9e9669b5c) |

## Testing Instructions
* Spin up a Jetpack Cloud live branch
* Pick a site with Jetpack VaultPress Backup and no credentials configured
* Navigate to Settings

Given the conditions could be difficult to reproduce, you could simulate sending the error on `/sites/
BLOG_ID/rewind/credentials/update` API. More details here: https://github.com/Automattic/jetpack-backup-team/issues/490#issuecomment-2111151691.

* Ensure each error described below has a `Learn more` link, and clicking on it opens a new tab with expected link and triggers a `calypso_jetpack_settings_credentials_error_learn_more_click` Tracks event. You can check this by looking at the `Network` tab and checking the t.gif requested when clicking the link.

## Pre-merge Checklist
